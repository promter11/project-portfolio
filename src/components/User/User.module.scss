@import "../../styles/_variables";

.user {
  text-align: center;
  color: $white-color;
}

.avatar {
  width: 10rem;
  height: 10rem;
  margin-bottom: 2rem;
  border-radius: 50%;
  object-fit: cover;
}

.title {
  font-size: 1.6rem;
  font-weight: 900;
  line-height: 150%;
  text-transform: uppercase;
}

.position {
  font-size: 1.2rem;
  font-weight: 100;
  line-height: 150%;
}
