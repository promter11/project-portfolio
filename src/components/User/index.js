import React from "react";
import { string } from "prop-types";

import Heading from "../Heading";

import styles from "./User.module.scss";

const User = ({ name, position, avatar }) => {
  return (
    <div className={styles.user}>
      <img className={styles.avatar} src={avatar} alt="Avatar" />
      <Heading className={styles.title} level={3}>
        {name}
      </Heading>
      <p className={styles.position}>{position}</p>
    </div>
  );
};

User.propTypes = {
  name: string.isRequired,
  position: string.isRequired,
  avatar: string.isRequired,
};

export default User;
