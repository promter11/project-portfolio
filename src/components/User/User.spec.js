import React from "react";
import { shallow } from "enzyme";

import User from "./index";

const props = {
  name: "John Doe",
  position: "QA Engineer",
  avatar: "../images/1",
};

describe("User component", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<User {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
