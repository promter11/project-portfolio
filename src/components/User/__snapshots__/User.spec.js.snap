// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders correctly with props 1`] = `
<div
  className="user"
>
  <img
    alt="Avatar"
    className="avatar"
    src="../images/1"
  />
  <Heading
    className="title"
    level={3}
  >
    John Doe
  </Heading>
  <p
    className="position"
  >
    QA Engineer
  </p>
</div>
`;
